pkg/services/ngalert/notifier/channels/kafka.go

```diff
@@ -102,6 +105,23 @@ func (kn *KafkaNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool,
 	ruleURL := joinUrlPath(kn.tmpl.ExternalURL.String(), "/alerting/list", kn.log)
 	bodyJSON.Set("client_url", ruleURL)
 
+	var imageURL string
+	_ = withStoredImages(ctx, kn.log, kn.images,
```

Comment:
1. I wonder if it makes sense to log the error.
2. If I read the closure correctly, the notification will have a screenshot of the last alert, which image URL was successfully resolved. Therefore, if there are many alerts in the group (and they can be from different rules, it will include a _randomly_ chosen screenshot. I see there two problems:
- we do not control the order of the alerts in the slice, and therefore there is no guarantee that we send the same screenshoot every time we notify user.
- alerts can belong to different rules (when user groups by some common label), which worsen the 2.1 because screenshots can be from different rules. So, there seems to be no guarantee that if at time X a user receives a screenshot from rule A, at time Y user will still receive a screenshot of rule A but not rule B.

This behavior at least should be documented here (https://grafana.com/docs/grafana/next/alerting/images-in-notifications/#supported-notifiers).  

3. Also, I think this behavior contradicts the way we calculate the state a few lines above https://github.com/grafana/grafana/pull/50758/files#diff-dea2e3e01b482e8394a6c087f5f7dcd4246480d2a401e9389d7b119b375a48e7R88-R92. 
`alerts.Status()` returns Alerting when the slice contains at least one alert that is firing. Therefore, there is a chance that we send a notification with status Alerting but include an image from the resolved state.

Reply:
We could also consider adding an additional context that includes all the image URLs and some metadata to connect the image to an alert, while leaving this existing context field in place.